tests/spark/test_spark_model_export.py

```diff
@@ -270,3 +289,14 @@ def _transform(self, dataset):
         mleap.save_model(spark_model=unsupported_model,
                          path=model_path,
                          sample_input=spark_model_iris.spark_df)
+
+
+def test_save_fails_with_sample_input_containing_unsupported_data_type(spark_context, model_path):
+    sql_context = SQLContext(spark_context)
+    unsupported_df = sql_context.createDataFrame([(1, "2016-09-30"), (2, "2017-02-27")])
+    unsupported_df = unsupported_df.withColumn("_2", unsupported_df._2.cast(DateType()))
+    pipeline = Pipeline(stages=[])
+    model = pipeline.fit(unsupported_df)
+    # The Spark `DateType` is not supported by MLeap, so we expect serialization to fail.
```

Comment:
Can we double-check some aspect of the raised exception in order to make sure it failed for the right reason? It could just be any arbitrary problem in the current code.